pkg/codegen/go/gen_program_expressions.go

```diff
@@ -1022,12 +1022,44 @@ func (nameInfo) Format(name string) string {
 	return name
 }
 
+// rewriteUnaryNegate rewrites UnaryOperation(-, Number(N)) into Number(-N).
+func rewriteUnaryNegate(expr model.Expression) (model.Expression, hcl.Diagnostics) {
```

Comment:
I think we should just lift this to pcl binding, not make it Go specific. I had to do the same thing for the other languages when adding bigint support and just did this simplification in bindUnaryOpExpression (https://github.com/pulumi/pulumi/pull/16939/files#diff-0925bf1716dbbe0ef4c7363db8d274d28213f80d7620a1a6a6213cd775c94bf5R786).

Feels like we may as well do that now, as it fixes this for Go and will be needed for bigints later anyway.

Reply:
Done ✅